### ISSUE TITLE
Tools get activated without pressing anything when using a tablet

### ISSUE THREAD

**Issue by Eth3reaI** [xournalpp/xournalpp#3274]
**Affects versions :**
 - OS: Linux - Manjaro (Arch)
 - Desktop environment: KDE - X11
 - libgtk: 3.24.30
 - Xournal++ version: 1.1.0
 - Installation method: official release

**Describe the bug**
I've tested this behavior using a Huion tablet with digimend kernel drivers, however it has never happened in previous versions.

When going over the toolbar with the cursor and then exiting and going back to the page, the selected tool gets activated without pressing anything. 

**To Reproduce**
1. Map the eraser on one of the tablet pen buttons
2. Select a tool (e.g. pen)
3. Draw something
4. Cancel what you just drew (using the mapped button)
5. Hover with the tablet pen cursor over the toolbar
6. Go back onto the page without touching the tablet, just hovering

Even if you're not touching the tablet with the pen it will start drawing at the minimum amount of pressure a very thin line.
As soon as you'll touch the tablet the wrong line will get deleted.

**Expected behavior**
The selected tool shouldn't be activated if I don't explicit press with the pen on the tablet.

**Screenshots of Problem**
Note: the thin lines are the ones where I'm not pressing anything, just hovering with the pen above the tablet.

![Alt Text](https://s6.gifyu.com/images/2021-07-26-14-08-19.gif)

**Additional context**
Tablet used: (Huion) Inspiroy h640p
Driver: digimend-kernel-drivers-dkms-git (AUR)

**Comment by rolandlo** [comment]
Do you have `Merge button events with stylus tip` option enabled in the input system panel in menu `Edit > Preferences` or not? It could make a difference. There have been other reports with thin lines like this and it will have to be fixed in #3039. Writing a stylus handler that works for everyone is tricky buisness, as drivers may have bugs and different drivers may have different bugs.

As a current workaround, you could try out setting `minimal pressure to 0.0` in the stylus panel in the preferences. Then the thin lines are not visible anymore, I think. They will still be stored (with pressure 0.0) in the `.xopp-file`, but at least they don't distort the picture.

**Comment by Eth3reaI** [comment]
@rolandlo Yes, I have `Merge button events with stylus tip` option enabled (without it the eraser wouldn't work). I will try with the workaround you suggested, thanks. However I've always been using the same tablet, settings and drivers even before, but I've experienced this problem just with the latest update. I suppose it's possible that maybe some changes in the codebase are now conflicting just with my specific tablet or drivers, but it could be useful to try to reproduce this behavior with other tablets/drivers as well, in order to verify if it's just a new bug or an instance of an older one already known.

**Comment by rolandlo** [comment]
This seems to be the same issue as #2762 and we have to find a solution for that problem.

**Comment by wtbgagoa** [comment]
Please check if the following commit fixes the problem:

```
$ git log -p
commit f354b62c615a15414bdceffe24a6708e8bcf9c41 (HEAD -> bugfix/#3274)
Author: wtbgagoa <wtbgagoa@gmail.com>
Date:   Sun Dec 12 19:46:43 2021 +0100

    Proposed fix for bug #3274.

diff --git a/src/core/gui/inputdevices/StylusInputHandler.cpp b/src/core/gui/inputdevices/StylusInputHandler.cpp
index db55a145..71aec3fd 100644
--- a/src/core/gui/inputdevices/StylusInputHandler.cpp
+++ b/src/core/gui/inputdevices/StylusInputHandler.cpp
@@ -158,17 +158,19 @@ void StylusInputHandler::setPressedState(InputEvent const& event) {
                 break;
             case 2:
                 this->modifier2 = false;
+                if (this->inputContext->getSettings()->getInputSystemTPCButtonEnabled()) {
+                   this->deviceClassPressed = false;
+                }
                 break;
             case 3:
                 this->modifier3 = false;
+                if (this->inputContext->getSettings()->getInputSystemTPCButtonEnabled()) {
+                   this->deviceClassPressed = false;
+                }
             default:
                 break;
         }
     }
-
-    if (this->inputContext->getSettings()->getInputSystemTPCButtonEnabled()) {
-        this->deviceClassPressed = this->deviceClassPressed || this->modifier2 || this->modifier3;
-    }
 }
 
```

**Comment by Technius** [comment]
Fixed by #3701.
